.github/workflows/ccpp.yml

```
@@ -49,3 +58,93 @@ jobs:
         make install
         ldd __pref/bin/ambs-lgp2-frontend
         __pref/bin/ambs-lgp2-frontend
+
+  build-msys2:
+    # only try expensive windows buildserver if linux build works
+    needs: build-ix
+
+    runs-on: windows-latest
+
+    # See https://github.com/msys2/setup-msys2
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { sys: mingw32, env: i686   }
+          - { sys: mingw64, env: x86_64 }
+          - { sys: ucrt64,  env: ucrt-x86_64 }
+          - { sys: clang64, env: clang-x86_64 }
+
+    name: 'MSYS2 ${{ matrix.sys }} build'
+
+    env:
+      MINGW_ARCH: ${{ matrix.sys }}
+
+    defaults:
+      run:
+        shell: msys2 {0}
+
+    steps:
+      - name: 'git config'
+        run: git config --global core.autocrlf input
+        shell: bash
+
+      - uses: actions/checkout@v3
+
+      - name: 'Set up MSYS2 and Install Software'
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{ matrix.sys }}
+          update: true
+          install: >-
+            git
+            autoconf
+            automake
+            gettext-devel
+            libtool
+            make
+            pkgconf
+            msys2-w32api-runtime
+            mingw-w64-${{ matrix.env }}-gcc
+            mingw-w64-${{ matrix.env }}-curl
+            mingw-w64-${{ matrix.env }}-libexif
+            mingw-w64-${{ matrix.env }}-libgd
+            mingw-w64-${{ matrix.env }}-libltdl
+            mingw-w64-${{ matrix.env }}-libusb
+            mingw-w64-${{ matrix.env }}-libxml2
+
+      - name: 'autoreconf'
+        run: autoreconf -vis .
+
+      - name: 'configure'
+        env:
+          CXX: no
+        run: ./configure ${COMMON_CONFIGURE_FLAGS} --prefix=$PWD/__prefix
+
+      - name: 'setup make with nproc'
+        run: NPROC=2; echo "MAKE=make -j$NPROC -l$NPROC" >> $GITHUB_ENV
```

maybe:

```suggestion
        run: NPROC=2 && echo "MAKE=make -j$NPROC -l$NPROC" >> $GITHUB_ENV
```

or (I don't think it's needed though)

```suggestion
        run: export NPROC=2 && echo "MAKE=make -j$NPROC -l$NPROC" >> $GITHUB_ENV
```